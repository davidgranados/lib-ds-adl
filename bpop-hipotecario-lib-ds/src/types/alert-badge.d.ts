import { ReactNode } from "react";
import { Testable } from "./global";

export interface AlertBadgeProps extends Testable {
  children?: ReactNode;
  type?: "info" | "warning" | "success" | "danger";
}
