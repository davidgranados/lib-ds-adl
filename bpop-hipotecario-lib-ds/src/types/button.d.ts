import { HTMLPropsFor } from "./component";

export type ButtonVariant = "primary" | "secondary";
export type ButtonTextSize = "sm" | "md" | "lg";
export type ButtonSize = "xs" | ButtonTextSize;

export interface DefaultButtonProps extends HTMLPropsFor<"button"> {
  label: string;
  variant?: ButtonVariant;
  disabled?: boolean;
  onClick?: () => void;
}

export interface ButtonProps extends DefaultButtonProps {
  size?: ButtonSize;
}

export interface ButtonTextProps extends DefaultButtonProps {
  size?: ButtonTextSize;
}
