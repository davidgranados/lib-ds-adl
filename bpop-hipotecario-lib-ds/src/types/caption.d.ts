import { ReactNode } from "react";
import { Testable } from "./global";

export declare type InputFieldCaptionStatus = "default" | "error" | "warning" | "success";

export declare interface InputFieldCaptionProps extends Testable {
  children: ReactNode;
  status?: InputFieldCaptionStatus;
}
