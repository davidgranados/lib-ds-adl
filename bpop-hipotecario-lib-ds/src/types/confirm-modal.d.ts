import { Testable } from "./global";

export declare interface ConfirmModalProps extends Testable {
  title: string;
  description?: string;
  show: boolean;
  mainButtonText?: string;
  secondaryButtonText?: string;
  onMainButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  onToggle?: (state: boolean) => void;
}
