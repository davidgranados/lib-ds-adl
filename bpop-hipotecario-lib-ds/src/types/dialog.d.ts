import { Testable } from "./global";

export declare interface DialogProps extends Testable {
  show: boolean;
  width?: number;
  onToggle?: (state: boolean) => void;
}
