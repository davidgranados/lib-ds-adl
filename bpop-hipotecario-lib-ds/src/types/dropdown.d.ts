import { Testable } from "./global";

export declare type DropdownItems = { label: string; value: string | number }[];
export declare type DropdownOnChange = (v: string | number) => void;
export declare type DropdownValue = string | number;

export declare interface DropdownProps extends Testable {
  label: string;
  items: DropdownItems;
  disabled?: boolean;
  hasError?: boolean;
  caption?: string;
  placeholder?: string;
  value?: DropdownValue;
  onChange: DropdownOnChange;
}
