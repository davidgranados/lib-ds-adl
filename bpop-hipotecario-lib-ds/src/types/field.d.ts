import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from "react";

import { InputBaseSize, InputBaseStatus } from "../../atoms/input-base";
import { TypographyStatus } from "../../atoms/typography";

export interface DefaultFieldProps {
  id: string;
  label: string;
  value?: string | number;
  status?: InputBaseStatus | TypographyStatus;
  caption?: string;
  placeholder?: string;
  size?: InputBaseSize;
  disabled?: boolean;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onFocus?: FocusEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  onInput?: FormEventHandler<HTMLInputElement>;
}
