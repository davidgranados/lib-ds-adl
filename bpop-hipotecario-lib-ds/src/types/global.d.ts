import React from "react";

export type HTMLPropsFor<Tag extends keyof JSX.IntrinsicElements> = JSX.IntrinsicElements[Tag];
export type CombinedRef = Array<React.MutableRefObject<null> | React.ForwardedRef<HTMLInputElement>>;
