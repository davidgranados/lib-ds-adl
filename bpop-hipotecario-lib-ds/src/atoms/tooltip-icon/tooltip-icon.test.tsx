import React from "react";
import { mount, ReactWrapper } from "enzyme";

import { TooltipIconProps } from "../../types";

import TooltipIcon from "./tooltip-icon";

describe("<TooltipIcon />", () => {
  let wrapper: ReactWrapper<typeof TooltipIcon>;
  const props: TooltipIconProps = {
    marginLeft: 20,
    marginRight: 20,
    textTitle: "tooltipTextTitle",
    textBody: "tooltipTextBody",
  };

  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof TooltipIcon>(<TooltipIcon {...props} />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
