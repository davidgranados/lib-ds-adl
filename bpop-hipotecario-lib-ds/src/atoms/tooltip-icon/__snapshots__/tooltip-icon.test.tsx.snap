// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TooltipIcon /> should match snapshot 1`] = `
Array [
  <span
    className="mobile-inline-only"
  >
    <div
      className="wrapper"
      data-testid="tooltip-wrapper"
      id="tooltip-wrapper"
      onClick={[Function]}
    >
      <div
        className="content-wrapper"
        data-testid="tooltip"
        id="tooltip"
        style={
          Object {
            "width": "calc(calc(100vw - 20px) - 20px)",
          }
        }
      >
        <div
          className="content-body"
          data-testid="tooltip-content-body"
          id="tooltip-content-body"
        >
          <span
            className="icon-wrapper"
            data-testid="tooltip-icon-wrapper"
            id="tooltip-icon-wrapper"
          >
            <span
              className="icon icon-close icon-close"
              data-testid="tooltip-close-icon"
              id="tooltip-close-icon"
            />
          </span>
          <div
            className="content-text"
            data-testid="tooltip-content-text"
            id="tooltip-content-text"
          >
            <span>
              tooltipTextTitle
            </span>
            <span>
              tooltipTextBody
            </span>
          </div>
        </div>
      </div>
      <span
        className="icon icon-exclamation-open-circle trigger"
        data-testid="icon-exclamation-open-circle"
        id="icon-exclamation-open-circle"
      />
    </div>
  </span>,
  <span
    className="desktop-inline-only"
  >
    <div
      className="wrapper"
      data-testid="tooltip-wrapper"
      id="tooltip-wrapper"
      onClick={[Function]}
    >
      <div
        className="content-wrapper content-wrapper--static-width"
        data-testid="tooltip"
        id="tooltip"
      >
        <div
          className="content-body"
          data-testid="tooltip-content-body"
          id="tooltip-content-body"
        >
          <span
            className="icon-wrapper"
            data-testid="tooltip-icon-wrapper"
            id="tooltip-icon-wrapper"
          >
            <span
              className="icon icon-close icon-close"
              data-testid="tooltip-close-icon"
              id="tooltip-close-icon"
            />
          </span>
          <div
            className="content-text"
            data-testid="tooltip-content-text"
            id="tooltip-content-text"
          >
            <span>
              tooltipTextTitle
            </span>
            <span>
              tooltipTextBody
            </span>
          </div>
        </div>
      </div>
      <span
        className="icon icon-exclamation-open-circle trigger"
        data-testid="icon-exclamation-open-circle"
        id="icon-exclamation-open-circle"
      />
    </div>
  </span>,
]
`;
